api/lib/mno_enterprise/concerns/controllers/jpi/v1/app_instances_controller.rb

```diff
@@ -13,31 +13,33 @@ module MnoEnterprise::Concerns::Controllers::Jpi::V1::AppInstancesController
   #==================================================================
   # Instance methods
   #==================================================================
-  # GET /mnoe/jpi/v1/organization/1/apps.json?timestamp=151452452345
+  # GET /mnoe/jpi/v1/organization/1/apps.json?tg=151452452345
   def index
-    @app_instances = parent_organization.app_instances.active.where("updated_at.gt" => Time.at(timestamp)).select do |i|
-      # force owner assignment to avoid a refetch in ability can?(:access,i)
-      i.owner = parent_organization
+    @app_instances = Mno::AppInstance.includes(:app, :owner).where(owner_id: parent_organization.id, status: Mno::AppInstance::ACTIVE_STATUSES).to_a.select do |i|
       can?(:access,i)
     end
   end
 
   # POST /mnoe/jpi/v1/organization/1/app_instances
   def create
     authorize! :manage_app_instances, parent_organization
-    app_instance = parent_organization.app_instances.create(product: params[:nid])
-    MnoEnterprise::EventLogger.info('app_add', current_user.id, 'App added', app_instance)
-    head :created
+    app_instance = parent_organization.provision_app_instance(params[:nid])
+    if app_instance.errors.any?
+      render json: app_instance.errors, status: :bad_request
+    else
+      MnoEnterprise::EventLogger.info('app_add', current_user.id, 'App added', app_instance.first)
+      head :created
+    end
   end
 
   # DELETE /mnoe/jpi/v1/app_instances/1
   def destroy
-    app_instance = MnoEnterprise::AppInstance.find(params[:id])
+    @app_instance = Mno::AppInstance.find_one(params[:id])
 
-    if app_instance
-      authorize! :manage_app_instances, app_instance.owner
-      MnoEnterprise::EventLogger.info('app_destroy', current_user.id, 'App destroyed', app_instance)
-      app_instance.terminate
+    if @app_instance
+      authorize! :manage_app_instances, @app_instance.owner
+      MnoEnterprise::EventLogger.info('app_destroy', current_user.id, 'App destroyed', @app_instance)
+      @app_instance = @app_instance.terminate.first
```

Comment:
I know, 

https://github.com/chingor13/json_api_client/issues/75

> This was intentional for a few reasons:
> 
> First, I always wanted to make a result set be available because there is extra meta data that comes back with the response. We could embed the meta data in a single resource object but that seems weird because it wouldn't be available on the resource objects fetched in a multi object response. I guess we could store a reference to the meta data but that also seems a bit janky.
> 
> Second, in the initial version of jsonapi, all responses returned arrays - even requests for a single resource. The rationale behind that was to make client side parsing easier - no matter what you were fetching, you could always iterate of the results like an array. There was a change (about a year ago?) to the spec that brought back the ability to respond with a single object or and array of objects as the main data. I wasn't a huge fan of it at the time.
> 
> Additionally, I really don't like the behavior in ActiveRecord where find can return a single result or an array depending on what input you give it. This departs from ActiveRecord but I'm ok with that here.
> 
> Hope this gives some insight into why it's doing this.